ansible/roles/gpu/tasks/nvidia-gpu.yaml

```diff
@@ -48,6 +48,58 @@
     path: "{{ nvidia_remote_bundle_path }}"
     state: directory
 
+- name: find nvidia driver installer file
+  find:
+    paths: "{{ nvidia_remote_bundle_path }}"
+    patterns: "NVIDIA-Linux-x86_64-*.run"
+  register: nvidia_driver_files
+
+- name: extract version from file names
+  set_fact:
+    nvidia_driver_versions: "{{ nvidia_driver_files.files | map(attribute='path') | map('regex_replace', '^.*NVIDIA-Linux-x86_64-(\\d+\\.\\d+\\.\\d+)\\.run$', '\\1') | list }}"
+    uninstall: false
+
+- name: compare versions with nvidia_driver_version
+  loop: "{{ nvidia_driver_versions }}"
+  when: item is version( nvidia_driver_version , '<')
+  set_fact:
+    uninstall: true
+    previous_version: "{{ item }}"
+
+- name: uninstall and stop programs using older nvidia drivers
+  block:
```

Comment:
Will this block only executed once for a last `previous_version` .  If intention is to remove all older versions, I think this block should be in the loop (not sure its possible)

Reply:
there should only be one older version at a time, but i pushed up some code to do it in a loop in case there are others

Reply:
Thanks faiq. it looks neat.